testsuite/tests/basic/equality.ml

```diff
@@ -72,35 +72,78 @@ let _ =
   test 36 eqm1 (compare (0.0, nan) (0.0, 0.0));
   test 37 eqm1 (compare (0.0, nan) (0.0, neg_infinity));
   test 38 eq0 (compare (nan, 0.0) (nan, 0.0));
-  let cmpgen x y = (x=y, x<>y, x<y, x<=y, x>y, x>=y) in
-  let cmpfloat (x:float) (y:float) = (x=y, x<>y, x<y, x<=y, x>y, x>=y) in
+
+  (* Force inline these functions to ensures that the simplifier will
```

Comment:
I would change this to say: "Force inline these functions to allow evaluation of the comparisons at compile time by the middle end".